sklearn/ensemble/_stacking.py

```diff
@@ -45,14 +45,20 @@
 def _estimator_has(attr):
     """Check if we can delegate a method to the underlying estimator.
 
-    First, we check the first fitted final estimator if available, otherwise we
-    check the unfitted final estimator.
+    First, we check the fitted `final_estimator_` if available, otherwise we check the
+    unfitted `final_estimator`. We raise the original `AttributeError` if `attr` does
+    not exist. This function is used together with `available_if`.
     """
-    return lambda self: (
-        hasattr(self.final_estimator_, attr)
-        if hasattr(self, "final_estimator_")
-        else hasattr(self.final_estimator, attr)
-    )
+
+    def check(self):
+        if hasattr(self, "final_estimator_"):
+            getattr(self.final_estimator_, attr)
+            return True
+        else:
+            getattr(self.final_estimator, attr)
+            return True
```

Comment:
nit, but don't need to return twice. Same for all the other instances of this method.

```suggestion
    def check(self):
        if hasattr(self, "final_estimator_"):
            getattr(self.final_estimator_, attr)
        else:
            getattr(self.final_estimator, attr)

        return True
```